microprofile.ls/org.eclipse.lsp4mp.ls/src/main/java/org/eclipse/lsp4mp/services/properties/PropertiesFileValidator.java

```diff
@@ -240,20 +240,21 @@ private void validatePropertyValueExpressions(String propertyName, ItemMetadata
 							}
 						} else {
 							if (propValExpr.hasDefaultValue()) {
+								// The expression has default value (ex : ${DBUSER:sa})
 								int start = propValExpr.getDefaultValueStartOffset();
 								int end = propValExpr.getDefaultValueEndOffset();
 								validatePropertyValue(propertyName, metadata, propValExpr.getDefaultValue(), start, end,
 									propValExpr.getOwnerModel());
 							} else {
-								// The expression has default value (ex : ${DBUSER:sa}) otherwise the error
-								// is reported
-								Range range = PositionUtils.createRange(propValExpr.getReferenceStartOffset(),
-									propValExpr.getReferenceEndOffset(), propValExpr.getDocument());
-								if (range != null) {
-									addDiagnostic("Unknown referenced property value expression '" + refdProp + "'",
-										range, expressionSeverity, ValidationType.expression.name());
-								} else {
-
+								if (!(Pattern.matches("[A-Z_]+[1-9A-Z_]*", refdProp.toString()))) {

```

Comment:
Try to avoid using Pattern which can be slow. If you need really using pattern , compile it at first. But in your case you don't need to use regexp. You can loop for characters of refProp and call `Character.isUpperCase`.

I suggest that you create a private static method  `isEnvVariable(String variable)` which does that.

Reply:
I used a pattern since `isUpperCase` simply checks character case, and misses validating special characters in ENVs. i.e. an ENV variable can include numbers and _, but cannot begin with numbers. Do you want me to make the validation less strict?